<?xml version="1.0"?>
<launch>
    <!-- Add brain node -->
    <node pkg="brain" type="hubert_brain" name="hubert_brain" output="screen">
        <param name="PAN_LB" value="30" />
        <param name="PAN_UB" value="150" />
        <param name="PAN_STEP_SIZE" value="15" />
        <param name="TILT_LB" value="70" />
        <param name="TILT_UB" value="110" />
        <param name="TILT_STEP_SIZE" value="10" />
        <param name="BODY_LB" value="60" />
        <param name="BODY_UB" value="120" />
        <param name="BODY_STEP_SIZE" value="2" />
    </node>

    <!-- Add Face tracker node -->
    <node type="trackface.py" pkg="brain" name="FaceTracker" >
        <param name="BODY_LB" value="60" />
        <param name="BODY_UB" value="120" />
    </node>

    <!-- Add Terminal application. Will run in a seperate terminal -->
    <node type="handleTerminal.py" pkg="brain" name="TerminalApplication" output="screen" launch-prefix="xterm -e"/>

    <!-- Add servo node (from ros_serial) -->
    <node type="serial_node.py" pkg="rosserial_python" name="ServoNode">
        <param name="port" value="/dev/ttyACM0" />
    </node>

    <!-- Language agent -->
    <node type="AgentMain.py" pkg="language" name="LanguageAgent" />



    <!-- Face detection and tracking-->
    <node pkg="face_detection" type="ros_face_detection" name="ros_face_detection" >
        <param name="SKIP_FRAMES" value="1" />
        <param name="RANGE_FOR_DETECTED" value="180" />
        <param name="RANGE_FOR_TRACKING" value="100" />
    </node>

    <!-- usb_cam launch-->
    <include file="$(find face_detection)/launch/usb_cam_.launch"/>
</launch>
